packages/analytics-browser-test/test/web-attribution.test.ts

```diff
@@ -357,12 +371,12 @@ describe('Web attribution', () => {
               client_upload_time: event_upload_time,
               events: [
                 generateAttributionEvent(++eventId, url),
-                generateSessionStartEvent(++eventId),
-                generatePageViewEvent(++eventId, 1, url),
-                generateSessionEndEvent(++eventId),
+                addPageUrlEnrichmentProperties(generateSessionStartEvent(++eventId), url),
+                generatePageViewEvent(++eventId, 1, url, undefined, { previousPageUrl: '' }),
+                addPageUrlEnrichmentProperties(generateSessionEndEvent(++eventId), directUrl, url),
```

Comment:
Tests are passing and make sense to me, except this, where session_end events have what is expected to be the next page's data.

When do session end events get fired?

Reply:
I see - that makes sense for why the events are showing up the way they are in tests then.